classes/constraints/isNotEmpty.php

```diff
@@ -3,27 +3,41 @@
 namespace mageekguy\atoum\phpunit\constraints;
 
 use mageekguy\atoum\asserters;
-use mageekguy\atoum\exceptions;
+use mageekguy\atoum\phpunit\constraint;
 use mageekguy\atoum\tools\variable\analyzer;
 
-class isNotEmpty extends notCount
+class isNotEmpty extends constraint
 {
+    private $analyzer;
+
     public function __construct($description = null, analyzer $analyzer = null)
     {
-        parent::__construct(0, $description, $analyzer);
+        $this->description = $description;
+        $this->analyzer = $analyzer ?: new analyzer();
     }
 
     protected function matches($actual)
     {
-        if ($this->analyzer->isString($actual)) {
-            $asserter = new asserters\phpString(null, $this->analyzer);
+        if ($actual instanceof \countable) {
+            $asserter = new asserters\sizeOf(null, $this->analyzer);
+            $asserter->setWith($actual)->isNotEqualTo(0, $this->description);
+        } else {
+            $asserter = new notEmptyAsserter(null, $this->analyzer);
             $asserter->setWith($actual)->isNotEmpty($this->description);
+        }
+    }
+}
+
+class notEmptyAsserter extends asserters\variable
```

Comment:
Please either:
 * extract this class to its own file
* include its code in the `isEmpty` constraint